yt_dlp/extractor/youtube.py

```diff
@@ -3307,7 +3306,57 @@ def _extract_heatmap(self, data):
                 'value': ('intensityScoreNormalized', {float_or_none}),
             })) or None
 
-    def _extract_comment(self, comment_renderer, parent=None):
+    def _extract_comment(self, view_model, entities, parent=None):
+        comment_entity_payload = traverse_obj(entities, (..., 'payload', 'commentEntityPayload', {dict}), get_all=False)
+        toolbar_entity_payload = traverse_obj(entities, (..., 'payload', 'engagementToolbarStateEntityPayload', {dict}), get_all=False)
+        comment_id = comment_entity_payload.get('properties').get('commentId')
+
+        info = {
+            'id': comment_id,
+            'text': try_get(comment_entity_payload, lambda x: x['properties']['content']['content'], str),
+            'like_count': parse_count(try_get(comment_entity_payload, lambda x: x['toolbar']['likeCountA11y'], str)) or 0,
+            'author_id': traverse_obj(comment_entity_payload, ('author', 'channelId', {self.ucid_or_none})),
+            'author': try_get(comment_entity_payload, lambda x: x['author']['displayName'], str),
+            'author_thumbnail': traverse_obj(comment_entity_payload, ('author', 'avatarThumbnailUrl', {url_or_none})),
+            'parent': parent or 'root',
```

Comment:
use a single traversal

```suggestion
            'parent': parent or 'root',
            **traverse_obj(comment_entity_payload, {
                'text': ('properties', 'content', 'content', {str}),
                'like_count': ('toolbar', 'likeCountA11y', {parse_count}),
                'author_id': ('author', 'channelId', {self.ucid_or_none}),
                'author': ('author', 'displayName', {str}),
                'author_thumbnail': ('author', 'avatarThumbnailUrl', {url_or_none}),
            }),
```

Reply:
also, in my WIP my branch I had the `author_thumbnail` traversal path as only `('avatarThumbnailUrl', {url_or_none})`. I was probably wrong, but could you double-check it please?

Reply:
Fixed, and I verified the ```avatarThumbnailUrl``` is correct.